### DISCUSSION TITLE
Java Agent Version adding for JAR

### DISCUSSION THREAD

**Discussion by AjinThankachan** [apache/skywalking#9144] [Q&A] [answered]
Hi,
I am using the latest Javaagent 8.10. Previously i was using the 8.9. I have put my queries below pleas help me to understand

1. When the upgrade is happening i wish to update the existing one without causing any issues in the existing connection is it possible.If so how. I mean i dont wanted to change my config.
2. In the java agent there is no version is mentioned along with  skywalking-agent.jar. When i am putting it on multiple system and to understand the version to provide frequent updates if using older is there any possibility to add like 'skywalking-agent_version.jar'. Is there any config or something to make these changes if so  where i can add the version variable or version number?
Can someone please help me so that my burden will be less. And i can provide the latest patches on timely manner with help of linux commands. Thanks in advance and i value your time and effort to do the good job.

**Comment by wu-sheng** [comment, marked as answer]
If you want to rename the jar file or folder name, do on your own, it is fine.
Officially, we don't have any plan to change the jar name in the release, because it is included in MENIFEST file, like Java suggested.

<img width="1380" alt="image" src="https://user-images.githubusercontent.com/5441976/170900332-b9ce893f-3589-4be4-a32a-017430afce79.png">

About (1), I don't know what is your mean about `existing connection`. Agent has been replaced with a JVM rebooting.

**Reply by AjinThankachan** [reply]
If i change the jar file name like skywalking-agent-8-9-1.jar is there any dependency issue or something. Just I need to change in the Javaagent path only na. Or do i need to do it while compiling the source project and do something. Can you please help me to understand that.Sorry, i am new to this hence asking. 
Thanks

**Reply by wu-sheng** [reply]
You could change the jar or the agent root folder names directly, nothing is related to compiling. But don't change the folder names inside the agent root folder, unless you could follow the documents to understand what does `mount` mean.

**Reply by AjinThankachan** [reply]
Thanks a lot let me try
